mitodl/micromasters
# Rework login prompts for backends

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #5044 
Fixes #5045 

#### What's this PR do?
- Moves the prompt to login with MITx Online into a react-based dialog
- Ensure the dialog appears when switching programs, including enrolling in a new one
- Add support for missing edX authentication too
- Fixes the label for the "View on edX" link to say "MITx Online" where appropriate

#### How should this be manually tested?

You should see the dialog under the following circumstances:
- Load the dashboard as a learner who is enrolled in an MITx Online program but has no MITx Online auth
- Switch from an edX program to an MITx Online one
- Enroll in an MITx Online program from the navbar program selector

You should **not** see the dialog if you load the dashboard for an edX-only program.

#### Screenshots
![Screen Shot 2021-09-16 at 15 48 26](https://user-images.githubusercontent.com/28598/133787473-599dade7-ca99-4022-93a4-d03dd5791a6d.png)
![Screenshot 2021-09-16 at 15-48-11 Learner Dashboard MITx MicroMasters](https://user-images.githubusercontent.com/28598/133787479-b3536849-a24c-428b-ae9c-cb0a57aa4257.png)
